astropy/table/tests/test_column.py

```diff
@@ -464,6 +465,28 @@ def test_mask_on_non_masked_table(self):
             t["a"].mask = [True, False]
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.array([1], dtype=np.int32),
+        [2.0],
+        ["x"],
+        [10] * u.m / u.s,
+        [object()],
+    ],
+)
+def test_deepcopy_object_column(data):
+    # see https://github.com/astropy/astropy/issues/13435
+    c1 = table.Column(data)
+    c2 = copy.deepcopy(c1)
+    assert c2 is not c1
+    assert c2[0] is not c1[0]
+
+    c3 = table.Column(c1, copy=True)
```

Comment:
So, if we go with my suggestion, for this case one would have `assert c3[0] is c1[0]` for the `object` column.